### PR TITLE
feat(suite): soften FW hash check for other-error

### PR DESCRIPTION
**Pull request by Lemonexe**
## Description

Create a softer version of `DeviceCompromised` modal specifically for FW hash check `'other-error'` to handle a false positive edge case (see issue)

:information_source: Designed by myslef, shall be reworded or tweaked with the product team.

## Related Issue

Resolve https://github.com/trezor/trezor-suite-private/issues/119

## Screenshots:

**Before:** Normal ghost modal
image

**After (only for `'other-error'`):** Softer ghost modal
image

### CODE REVIEW

File: packages/suite/src/components/suite/SecurityCheck/SecurityCheckFail.tsx [hunk 1]
```diff
@@ -19,33 +21,20 @@ const Flex = styled.div`
...
-    },
-    {
-        icon: 'hand',
-        content: <Translation id="TR_AVOID_USING_DEVICE" />,
-    },
-    {
-        icon: 'chat',
-        content: <Translation id="TR_USE_CHAT" values={{ b: chunks => <b>{chunks}</b> }} />,
-    },
-] as const;
```
Comment (Lemonexe) [inline packages/suite/src/components/suite/SecurityCheck/SecurityCheckFail.tsx:35]: Moved to a new file for brevity

File: packages/suite/src/components/suite/SecurityCheck/DeviceCompromised.tsx [hunk 2]
```diff
@@ -48,12 +49,22 @@ export const DeviceCompromised = () => {
         if (hashCheckError) reportCheckFail('hash', { ...commonData, hashCheckError });
     }, [revisionCheckError, hashCheckError, revision, vendor, version]);
 
+    const isSoftFailure = revisionCheckError === null && hashCheckError === 'other-error';
```
Comment (Lemonexe) [inline packages/suite/src/components/suite/SecurityCheck/DeviceCompromised.tsx:52]: A countefeit device could deliberately crash hash check plus fail revision check, so if revision check did not pass, it has precedence (:arrow_right: display normal ghost modal)

File: packages/suite/src/components/suite/SecurityCheck/checklistItems.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Translation } from '../Translation';
+
+export const hardFailureChecklistItems = [
```
Comment (peter-sanderson) [inline packages/suite/src/components/suite/SecurityCheck/checklistItems.tsx]: Maybe declarative typing instead of `as const`?

Comment (peter-sanderson) [reply packages/suite/src/components/suite/SecurityCheck/checklistItems.tsx]: `: SecurityChecklistItem[]`?

**Review by peter-sanderson — APPROVED** [review]

File: packages/suite/src/components/suite/SecurityCheck/checklistItems.tsx (hunk 3 above)
Comment (Lemonexe) [reply packages/suite/src/components/suite/SecurityCheck/checklistItems.tsx]: oh yeah, fixed. As we discussed yesterday the use-cases for `as const`, this really isn't one, I mindlessly copied old code..

**Pull request merged**